### ISSUE TITLE
Fix and update softmax derivation backprop formula

### ISSUE THREAD

**Issue by daviddwlee84** [daviddwlee84/DeepLearningPractice#1]
https://github.com/ckyken/DL/tree/master/HW1
